Backend/database/repositories/filesystem.go

```diff
@@ -3,14 +3,87 @@ package repositories
 
 import (
 	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
 	"strings"
+
+	"github.com/docker/docker/client"
 )
 
+const volume_path = "/var/lib/documents/data"
+
 // Implements IRepositoryInterface
 type FilesystemRepository struct {
 	embeddedContext
 }
 
+type DockerFileystemRepository struct {
+	cli *client.Client
+}
+
+// Create instance of DockerFileSystemRepository struct
+func NewDockerFilesystemRespository() (c *DockerFileystemRepository, err error) {
+	c = new(DockerFileystemRepository)
+
+	c.cli, err = client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return nil, err
+	}
+	return c, nil
```

Comment:
```suggestion
    return client.NewClientWithOpts(client.FromEnv)
```
Is there a purpose for this if statement?

- If err is not nil, then we convert the `*DockerFilesystemRepository` to `nil` in the return value and return error as it already is (unlike other cases where you seem to change the error message)
  - This is not required as if the error handling is implemented correctly when this function is called, we wouldn't do anything with the `*DockerFilesystemRepository` when we catch the error
- If the err is nil, then we return `nil` instead of the error and return c normally
  - This doesn't do anything since the error is already nil

Hence you should be able to return the values as is.